controllers/productController.js

```diff
@@ -0,0 +1,33 @@
+const productService = require("../services/productService");
+
+const allProducts = async (req, res) => {
+  const pageInfo = req.query;
+  const page = parseInt(pageInfo.page);
+  const pageSize = parseInt(pageInfo.pageSize);
+  const category = pageInfo.category;
+  const sub = pageInfo.sub;
+  const ty = pageInfo.type;
+  const type = ty ? ty.split(",") : "";
+  const sort = pageInfo.sort;
```

Comment:
- 4~6번 라인의 코드는 페이지네이션 구현을 위한 query parameter로 보입니다.
페이지네이션을 구현할 때 LIMIT 구문을 잘 사용해주셨는데요, LIMIT과 OFFSET에 해당하는 데이터는 pageInfo라는 객체에 별도의 key: value 형태로 안받아도 될 것 같습니다!
프론트엔드 개발자들과 다시 논의하여 limit과 offset이라는 key에 페이지에 대한 number 데이터만 받아와도 될 것 같습니다.

- 나머지 데이터 구조분해할당을 통해 데이터를 받아오면 조금 더 간결하게 코드를 작성할 수 있을 것으로 보입니다.
아래 예시 코드를 참고해주세요!

 ```suggestion
  const { category, subCategory, type, sort, limit, offset } = req.query
  const productType = type ? type.split(",") ; ""; // 해당 코드는 service 단에서 다루는 것이 좋을 것 같습니다!
```